pytroll-exercise-2018-10/5-custom-reader.ipynb

```diff
@@ -0,0 +1,383 @@
+    "    calibration:\n",
+    "      reflectance:\n",
+    "        standard_name: toa_bidirectional_reflectance\n",
+    "        units: \"%\"\n",
+    "      radiance:\n",
+    "        standard_name: toa_outgoing_radiance_per_unit_wavelength\n",
+    "        units: W m-2 um-1 sr-1\n",
+    "      counts:\n",
+    "        standard_name: counts\n",
+    "        units: count\n",
+    "    file_type: nc_seviri_l1b\n",
+    "\n",
+    "  VIS008:\n",
+    "    name: VIS008\n",
+    "    resolution: 3000.403165817\n",
+    "    wavelength: [0.74, 0.81, 0.88]\n",
+    "    calibration:\n",
+    "      reflectance:\n",
+    "        standard_name: toa_bidirectional_reflectance\n",
+    "        units: \"%\"\n",
+    "      radiance:\n",
+    "        standard_name: toa_outgoing_radiance_per_unit_wavelength\n",
+    "        units: W m-2 um-1 sr-1\n",
+    "      counts:\n",
+    "        standard_name: counts\n",
+    "        units: count\n",
+    "    file_type: nc_seviri_l1b\n",
+    "\n",
+    "  WV_062:\n",
+    "    name: WV_062\n",
+    "    resolution: 3000.403165817\n",
+    "    wavelength: [5.35, 6.25, 7.15]\n",
+    "    calibration:\n",
+    "      brightness_temperature:\n",
+    "        standard_name: toa_brightness_temperature\n",
+    "        units: \"K\"\n",
+    "      radiance:\n",
+    "        standard_name: toa_outgoing_radiance_per_unit_wavelength\n",
+    "        units: W m-2 um-1 sr-1\n",
+    "      counts:\n",
+    "        standard_name: counts\n",
+    "        units: count\n",
+    "    file_type: nc_seviri_l1b\n",
+    "\n",
+    "  WV_073:\n",
+    "    name: WV_073\n",
+    "    resolution: 3000.403165817\n",
+    "    wavelength: [6.85, 7.35, 7.85]\n",
+    "    calibration:\n",
+    "      brightness_temperature:\n",
+    "        standard_name: toa_brightness_temperature\n",
+    "        units: \"K\"\n",
+    "      radiance:\n",
+    "        standard_name: toa_outgoing_radiance_per_unit_wavelength\n",
+    "        units: W m-2 um-1 sr-1\n",
+    "      counts:\n",
+    "        standard_name: counts\n",
+    "        units: count\n",
+    "    file_type: nc_seviri_l1b\n",
+    "\n",
+    "    \n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The YAML file is now ready, let's go on with the corresponding python file.\n",
+    "## The python file\n",
+    "The python files needs to implement a file handler class for each file type that we want to read.\n",
+    "Such a class needs to implement a few methods:\n",
+    "- the `__init__` method, that takes as arguments the filename, the filename info that we get by parsing the filename using the pattern defined in the yaml file, and the filetype info that we get from the filetype definition in the yaml file. This method can also recieve other file handler instances as parameter if the filetype at hand has requirements. (See the explanation in the YAML file filetype section above)\n",
```

Comment:
"the filename info that we get by parsing the filename using the pattern defined in the yaml file". Can you put "dictionary" in there somewhere?

Reply:
should be "receive"